### PR TITLE
[Windows] add process agent support

### PR DESCRIPTION
**Pull request by derekwbrown**
### What does this PR do?

Adds process agent support.

### CODE REVIEW

**Review by ofek — CHANGES_REQUESTED** [review]

File: omnibus/config/software/datadog-process-agent-win.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    }
+  else
+    env = {
+        'GOPATH' => gopath.to_path,
+        'PATH' => "#{gopath.to_path}/bin:#{ENV['PATH']}",
+        'TRACE_AGENT_VERSION' => default_version, # used by gorake.rb in the trace-agent
+    }
+  end
+
+  #command "go get github.com/Masterminds/glide", :env => env
```
[the remark this answers is not in the document]
Comment (derekwbrown) [reply omnibus/config/software/datadog-process-agent-win.rb]: yes.  Gets a newer version of glide, which doesn't work on windows :(.  Fortunately, we're moving to dep.

**Review by ofek — DISMISSED** [review]

File: omnibus/config/projects/agent.rb [hunk 2]
```diff
@@ -128,7 +128,7 @@
 
 # External agents
 dependency 'datadog-trace-agent'
-if linux?
+if not osx?
```
Comment (gmmeyer) [inline omnibus/config/projects/agent.rb]: Not a blocker but idiomatic ruby uses `!` instead of `not`

**Review by gmmeyer — DISMISSED** [review]
🚢 👍

**Review by gmmeyer — DISMISSED** [review]
:shipit:

**Review by truthbk — APPROVED** [review]
Looks good! 🚀

**Pull request merged**